blockchain/genesis.go

```diff
@@ -255,7 +255,7 @@ func SetupGenesisBlock(db database.DBManager, genesis *Genesis, networkId uint64
 	// Special case: don't change the existing config of a non-mainnet chain if no new
 	// config is supplied. These chains would get AllProtocolChanges (and a compat error)
 	// if we just continued here.
-	if genesis == nil && stored != params.CypressGenesisHash {
+	if genesis == nil && params.CypressGenesisHash != stored && params.BaobabGenesisHash != stored {
```

Comment:
Could you know why go-eth decided to check only the mainnet hash?
go-eth code : 
```
	// Special case: don't change the existing config of a non-mainnet chain if no new
	// config is supplied. These chains would get AllProtocolChanges (and a compat error)
	// if we just continued here.
	if genesis == nil && stored != params.MainnetGenesisHash {
		return storedcfg, stored, nil
	}
```

Reply:
@winnie-byun Oh, it's really good question. If you look at SetEthKlay, you can see that mainnet flag support is recent. https://github.com/ethereum/go-ethereum/pull/21932/commits/ae00010b95b3c6c92eb34d55bdee139b54ef7f30. However, above code is written at 2017. I think that checking the mainnetGenesisHash was necessary in the past, but it seems to be unnecessary now as the  https://github.com/ethereum/go-ethereum/pull/21932/commits/ae00010b95b3c6c92eb34d55bdee139b54ef7f30 enters. (when mainnet, genesis != not nil) What do you think?
Also, I did some tests about changing hard fork block number in the mainnet and private network without `sored != params.MainnetGenesisHash`, the same result came out :)

Reply:
```suggestion
	if genesis == nil && (params.CypressGenesisHash != stored || params.BaobabGenesisHash != stored) {
```